apollo-router/src/services/connector_service.rs

```diff
@@ -269,13 +307,22 @@ impl ServiceFactory<ConnectRequest> for ConnectorServiceFactory {
     type Service = BoxService;
 
     fn create(&self) -> Self::Service {
-        ConnectorService {
-            _schema: self.schema.clone(),
-            _subgraph_schemas: self.subgraph_schemas.clone(),
-            _subscription_config: self.subscription_config.clone(),
-            connectors_by_service_name: self.connectors_by_service_name.clone(),
-            connector_request_service_factory: self.connector_request_service_factory.clone(),
-        }
+        // For backward compatibility, create a service that delegates to the specific services
+        // This is a bit of a hack, but needed for tests
+        let services = self.services.clone();
+
+        tower::service_fn(move |request: ConnectRequest| {
+            let services = services.clone();
+            let service_name = request.service_name.to_string();
+
+            async move {
+                if let Some(mut service) = services.get(&service_name).cloned() {
+                    service.call(request).await
```

Comment:
I think we should always call `.ready().await` before the call.

Reply:
I'm concerned it might break back pressure

Reply:
is it "not needed" because of a refactor? i can't find a .call in the PR anymore which is one way to solve this. (when you do use `.call()`, it absolutely is required to ready the service first, even if it appears to work in practice without doing so)